### DISCUSSION TITLE
Now Playing sync with audio

### DISCUSSION THREAD

**Discussion by eranwo** [AzuraCast/AzuraCast#4064] [Q&A]
**Using Docker installation method**
 Yes

**AzuraCast version**
 Rolling Release #9af25b3 (2021-03-09 8:49)

**Host Operating System**
Ubuntu 20.04

**Describe the bug**
When using a relay server there is a delay created between `Now playing` data and the audio streaming. The  `nowplaying`  API data , is ahead of the audio for about 20-30 sec
Is there a way to delay the `nowplaying` API data in order to get better sync with the audio stream especially when using external relay servers?

**Comment by appnostic-io** [comment]
It's a little tricky and an almost uphil battle due to the various ways clients and servers apply buffering.  The more (or less) buffering in the chain the more the time delay varies each way.  Moreover as an API which only offers data when called there is an additional delay presented since there's no guarantee that the API call and the start of the audio stream will sync.

Whilst one may eventually get to some sort of happy medium in AutoDJ the problem presents itself again during any type of live DJ or remote streams.

Just about the only reliable way for this metadata is inside the streams because these are coded and embedded alongside the audio but there's only a handful of web players that do this depending on their external libraries.
